pkg/controller/elasticsearch/filesettings/file_settings.go

```diff
@@ -0,0 +1,164 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package filesettings
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	policyv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/stackconfigpolicy/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/hash"
+)
+
+const defaultCompatibilityVersion = "8.5.0"
+
+// Settings represents the "File-based Settings" to write to the JSON file watched by Elasticsearch.
+type Settings struct {
+	Metadata SettingsMetadata `json:"metadata"`
+	State    SettingsState    `json:"state"`
+}
+
+// SettingsMetadata represents the metadata of the "File-based Settings".
+// Settings are versioned and any change in the Settings state must be followed by a version increase.
+type SettingsMetadata struct {
+	Version       string `json:"version"`
+	Compatibility string `json:"compatibility"`
+}
+
+// SettingsState represents the state of the "File-based Settings".
+// This is where the configuration of Elasticsearch objects resides.
+type SettingsState struct {
+	ClusterSettings      *commonv1.Config `json:"cluster_settings,omitempty"`
+	SnapshotRepositories *commonv1.Config `json:"snapshot_repositories,omitempty"`
+	SLM                  *commonv1.Config `json:"slm,omitempty"`
+	// RoleMappings           *commonv1.Config `json:"role_mappings,omitempty"`
+	// Autoscaling            *commonv1.Config `json:"autoscaling,omitempty"`
+	// IndexLifecyclePolicies *commonv1.Config `json:"ilm,omitempty"`
+	// IngestPipelines        *commonv1.Config `json:"ingest_pipelines,omitempty"`
+	// IndexTemplates         *IndexTemplates  `json:"index_templates,omitempty"`
+}
+
+// type IndexTemplates struct {
+//	ComponentTemplates       *commonv1.Config `json:"component_templates,omitempty"`
+//	ComposableIndexTemplates *commonv1.Config `json:"composable_index_templates,omitempty"`
+//}
+
+// hash returns the hash of the Settings, considering only the State without the Metadata.
+func (s *Settings) hash() string {
+	return hash.HashObject(s.State)
+}
+
+// NewSettings returns empty new Settings with their version created from the epoch time.
+func NewSettings(version int64) (int64, Settings) {
```

Comment:
nit, makes it easier to read the code
```suggestion
func NewEmptySettings(version int64) (int64, Settings) {
```